cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -24,35 +34,98 @@
 
     private final Vertx vertx;
     private final long idleTimeout;
-
-    public CruiseControlApiImpl(Vertx vertx) {
-        this(vertx, HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS);
+    private Secret ccSecret;
+    private Secret ccApiSecret;
+    private Secret coSecret;
+    private boolean apiAuthorizationEnabled;
+    private boolean apiAuthenticationEnabled;
+    private HTTPHeader authHttpHeader;
+
+    public CruiseControlApiImpl(Vertx vertx, Secret ccSecret, Secret ccApiSecret, Secret coSecret, Boolean apiAuthorizationEnabled, boolean apiAuthenticationEnabled) {
+        this.vertx = vertx;
+        this.idleTimeout = HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS;
+        this.ccSecret = ccSecret;
+        this.ccApiSecret = ccApiSecret;
+        this.coSecret = coSecret;
+        this.apiAuthorizationEnabled = apiAuthorizationEnabled;
+        this.apiAuthenticationEnabled = apiAuthenticationEnabled;
+        this.authHttpHeader = getAuthHttpHeader();
     }
 
-    public CruiseControlApiImpl(Vertx vertx, int idleTimeout) {
+    public CruiseControlApiImpl(Vertx vertx, int idleTimeout, Secret ccSecret, Secret ccApiSecret, Secret coSecret, Boolean apiAuthorizationEnabled, boolean apiAuthenticationEnabled) {
         this.vertx = vertx;
         this.idleTimeout = idleTimeout;
+        this.ccSecret = ccSecret;
+        this.ccApiSecret = ccApiSecret;
+        this.coSecret = coSecret;
+        this.apiAuthorizationEnabled = apiAuthorizationEnabled;
+        this.apiAuthenticationEnabled = apiAuthenticationEnabled;
+        this.authHttpHeader = getAuthHttpHeader();
     }
 
     @Override
     public Future<CruiseControlResponse> getCruiseControlState(String host, int port, boolean verbose) {
         return getCruiseControlState(host, port, verbose, null);
     }
 
+    public HttpClientOptions getHttpClientOptions() {
+
+        if (apiAuthenticationEnabled) {
+            String trustStorePassword = null;
+            File truststoreFile = null;
+
+            PasswordGenerator pg = new PasswordGenerator(12);
+            trustStorePassword = pg.generate();
+            truststoreFile = Util.createFileTrustStore(getClass().getName(), "ts", Ca.cert(ccSecret, "cruise-control.crt"), trustStorePassword.toCharArray());
```

Comment:
This method is called for every client request. So we are recreating this trust store file and password for every single request? Is that strictly required? Can this not be done once when the client is created? Or is there some security thing I am missing?

Reply:
Good call, I've moved this logic to the constructor so that it is only instantiated once per Rebalance reconcile